### PR TITLE
Update stackdriver plugins to reflect new product naming

### PR DESCRIPTION
**Pull request by danielnelson**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/stackdriver/README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Stackdriver Input Plugin
```
Comment (reimda) [inline plugins/inputs/stackdriver/README.md]: Heading should be "Google Cloud Monitoring Input Plugin"

File: plugins/outputs/stackdriver/README.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
 # Stackdriver Output Plugin
```
Comment (reimda) [inline plugins/outputs/stackdriver/README.md]: Heading should be "Google Cloud Monitoring Output Plugin"

**Comment by danielnelson** [comment]
It's possible we will want to add a plugin alias in the future, similar to with http_listener/influxdb_listener.  We could name it `cloud_monitoring` to match `cloud_pubsub`.  For now though, I think we should leave it primarily as `stackdriver` since I think this is the name with the most recognition.

**Comment by danielnelson** [comment]
Updated the plugin readme titles to, for example, `Stackdriver Google Cloud Monitoring Input Plugin`.

**Comment by uday1984** [comment]
"Google Cloud Monitoring Input Plugin" 
Can you share the definition to be define on the telegraf.conf file for the metric label and resource label 
I defined the conf file as below and verifed the data is appearing in GoogleCloudPlafform. However telegraf is not able to read it.

**filter: "metric.type = "compute.googleapis.com/instance/cpu/usage_time"" AND "metric.labels.instance_name = "demo-instance-1""**

ERROR

2021-09-17T12:43:11Z E! [inputs.stackdriver] Failed iterating time series responses: "name:\"projects/cfs-practice\"  filter:\"metric.type = \\\"compute.googleapis.com/quota/subnet_ranges_per_vpc_network/exceeded\\\" AND resource.labels. = \\\"\\\" AND metric.labels. = \\\"\\\"\"  interval:{end_time:{seconds:1631882284}  start_time:{seconds:1631882224}}": rpc error: code = InvalidArgument desc = Field filter had an invalid value of "metric.type = "compute.googleapis.com/quota/subnet_ranges_per_vpc_network/exceeded" AND resource.labels. = "" AND metric.labels. = """: Could not parse filter "metric.type = \"compute.googleapis.com/quota/subnet_ranges_per_vpc_network/exceeded\" AND resource.labels. = \"\" AND metric.labels. = \"\""; syntax error at line 1, column 104, token ' ';

**Pull request merged**